### PR TITLE
feat: cleanup PodPriority features gate

### LINKED ISSUE #77152: cleanup features.PodPriority
As PodPriority has been GA in 1.14, and we've removed FIFO implementation in scheduler.

I think it'd good to remove code related with "checking if features.PodPriority feature gate is enabled or not" - as that's always true now.

/sig scheduling
/kind cleanup

### PR DESCRIPTION
**Pull request by draveness**
/kind cleanup

**What this PR does / why we need it**:

PodPriority has already graduated to GA, and we could clean up the PodPriority gate which always true.

**Which issue(s) this PR fixes**:

Fixes https://github.com/kubernetes/kubernetes/issues/77152

**Special notes for your reviewer**:

I'm not sure about the release note, feel free to leave comments on that.

**Does this PR introduce a user-facing change?**:

```release-note
The GA PodPriority feature gate is now on by default and cannot be disabled. The feature gate will be removed in v1.18.
```

### CODE REVIEW

**Review by mattjmcnaughton — APPROVED** [review]
/lgtm

I'm not 100% up to date on the process for cleaning up a feature flag, but on the surface, this makes sense to me. All the code changes look safe.

Thanks for this cleanup.

File: pkg/features/kube_features.go [hunk 1]
```diff
@@ -509,7 +509,7 @@ var defaultKubernetesFeatureGates = map[featuregate.Feature]featuregate.FeatureS
 	Sysctls:                                     {Default: true, PreRelease: featuregate.Beta},
 	DebugContainers:                             {Default: false, PreRelease: featuregate.Alpha},
 	PodShareProcessNamespace:                    {Default: true, PreRelease: featuregate.Beta},
-	PodPriority:                                 {Default: true, PreRelease: featuregate.GA},
+	PodPriority:                                 {Default: true, PreRelease: featuregate.GA, LockToDefault: true},
```
Comment (liggitt) [inline pkg/features/kube_features.go]: add a comment indicating what release to remove this (v1.18)

File: pkg/kubelet/eviction/helpers_test.go [hunk 2]
```diff
@@ -719,8 +691,6 @@ func TestOrderedbyDiskDisableLocalStorage(t *testing.T) {
 }
 
 func TestOrderedbyInodes(t *testing.T) {
-	defer featuregatetesting.SetFeatureGateDuringTest(t, utilfeature.DefaultFeatureGate, features.PodPriority, true)()
-	defer featuregatetesting.SetFeatureGateDuringTest(t, utilfeature.DefaultFeatureGate, features.LocalStorageCapacityIsolation, true)()
```
Comment (liggitt) [inline pkg/kubelet/eviction/helpers_test.go]: was removing LocalStorageCapacityIsolation accidental?

**Comment by liggitt** [comment]
/approve

two nits, lgtm otherwise

**Comment by draveness** [comment]
> /approve
> 
> two nits, lgtm otherwise

Comments have been addressed, please take another look, thanks @liggitt

**Comment by draveness** [comment]
> /lgtm

It didn't work 🤣

**Pull request merged**